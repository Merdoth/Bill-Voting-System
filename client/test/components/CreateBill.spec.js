import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { CreateBill } from
  '../../js/components/CreateBill';

const props = {
  user: {
    token: {}
  },
  createBillRequest: jest.fn(),
};

/* global jest */
const setup = () => shallow(<CreateBill {...props} />);

describe('CreateBill Component snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('onSubmit()', () => {
  it('should add recipes to the state', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    const wrapper = setup();
    const form = wrapper.find('form');
    form.simulate('submit', event);
    expect(props.createBillRequest).toHaveBeenCalled();
  });
});

describe('onChange()', () => {
  it('should set title to state when input values changes', () => {
    const event = {
      target: { name: 'title', value: '' }
    };
    const wrapper = setup();
    const titleInput = wrapper.find('#title');

    event.target.value = 'Seyi Law in senate';
    titleInput.simulate('change', event);

    expect(wrapper.instance().state.title).toBe('Seyi Law in senate');
  });

  it('should set description state when input values changes', () => {
    const event = {
      target: { name: 'description', value: '' }
    };
    const wrapper = setup();
    const descriptionInput = wrapper.find('#description');

    event.target.value = 'This is how nigeria senate shows its madness';
    descriptionInput.simulate('change', event);

    expect(wrapper.instance().state.description)
      .toBe('This is how nigeria senate shows its madness');
  });
});
