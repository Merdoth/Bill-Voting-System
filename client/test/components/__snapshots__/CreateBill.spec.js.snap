// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateBill Component snapshot it should render the right amount of elements 1`] = `
<div
  className="dashboard-container"
>
  <Connect(SideBar) />
  <main>
    <div
      className="welcome-caption"
    >
      <div
        className="mobile-hambuger"
      >
        <a
          className="button-collapse hide-on-large-only "
          data-activates="slide-out"
          href="#"
        >
          <i
            className="material-icons"
          >
            menu
          </i>
        </a>
      </div>
      <span
        className="float-header-with-flex"
      />
      <span>
        <h5
          className="username-format"
        >
          <i
            className="material-icons username-icon"
          >
            account_circle
          </i>
          Welcome, 
        </h5>
      </span>
    </div>
    <div
      className="container"
    >
      <div
        className="edit-wrapper"
      >
        <div>
          <h1>
            Create a bill
          </h1>
        </div>
        <div
          className="account-summary-section"
        >
          <div>
            <form
              className="col s12 create"
              onSubmit={[Function]}
            >
              <div
                className="row"
              >
                <div
                  className="input-field col s12 m12"
                >
                  <input
                    className="form-control"
                    id="title"
                    name="title"
                    onChange={[Function]}
                    pattern=".{5,50}"
                    required={true}
                    title="5 characters minimum"
                    type="text"
                    value=""
                  />
                  <label
                    className="control-label"
                    htmlFor="title"
                  >
                    Title :
                  </label>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="input-field col s12 m12"
                >
                  <input
                    className="form-control"
                    id="description"
                    name="description"
                    onChange={[Function]}
                    pattern=".{5,300}"
                    required={true}
                    title="5 characters minimum"
                    type="text"
                    value=""
                  />
                  <label
                    className="control-label"
                    htmlFor="description"
                  >
                    Description :
                  </label>
                </div>
              </div>
              <div
                className="row form-cta"
              >
                <button
                  className="btn shadow-effect"
                  href="#"
                  type="submit"
                >
                  Create
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
